### PR TITLE
Add linux-specific `core.sys.linux.file` module

### PR DESCRIPTION
**Pull request by Burgos**
This adds core.sys.linux.file module which contains
flock function definition and values of the parameters
to pass.

### CODE REVIEW

**Comment by CyberShadow** [comment]
I believe this should be `core.sys.linux.sys.file` to follow the naming scheme for `sys/file.h`.

**Comment by CyberShadow** [comment]
BTW, unless you specifically need non-advisory locks, we already have definitions for fcntl locks, also exposed via `std.stdio.File.lock()` etc.

**Comment by CyberShadow** [comment]
Makefiles and file lists need to be modified when adding new modules. See all occurrences of an existing module name for an example, e.g. `git grep timerfd | grep -v ^src/`.

**Comment by Burgos** [comment]
> I believe this should be core.sys.linux.sys.file to follow the naming scheme for sys/file.h.

Thanks, I got fooled by the `sys` prefix.

Updated with moved location and updated `make/COPY`, `win32.mak` and `win64.mak` files.

File: src/core/sys/linux/sys/file.d [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+/* Operations for the `flock' call. */
+/// Shared lock
+enum LOCK_SH              = 0x01;
+/// Exclusive lock
+enum LOCK_EX              = 0x02;
+/// Unlock
+enum LOCK_UN              = 0x08;
+/// Atomic update
+enum __LOCK_ATOMIC        = 0x16;
```
Comment (CyberShadow) [inline src/core/sys/linux/sys/file.d]: This one isn't in my `file.h`. Is it Debian-specific?

Comment (CyberShadow) [inline src/core/sys/linux/sys/file.d]: I think this one is only implemented in the GNU Hurd kernel.

Comment (Burgos) [inline src/core/sys/linux/sys/file.d]: Hmm, I got it from mine (Ubuntu's) file.h, but I don't see it on glibc's upstream file.h, so I'll remove it.

Thanks and good catch!

**Comment by Burgos** [comment]
Updated with `__LOCK_ATOMIC` removed.

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Comment by DmitryOlshansky** [comment]
Looks good apart from horrible windows makefiles... but that's separate problem.

**Comment by andralex** [comment]
thx!

**Pull request merged**